flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/optimize/JoinHintResolver.java

```diff
@@ -172,21 +187,49 @@ && matchIdentifier(option, rightName.get())) {
     private void validateHints() {
         Set<RelHint> invalidHints = new HashSet<>(allHints);
         invalidHints.removeAll(validHints);
+        String errorPattern;
+
+        // firstly, check the unknown table (view) names in hints
+        errorPattern =
+                "The options of following hints cannot match the name of "
+                        + "input tables or views: %s";
+        StringBuilder errorMsgSb = new StringBuilder();
+        AtomicBoolean containsInvalidOptions = new AtomicBoolean(false);
```

Comment:
why AtomicBoolean needed here?

Reply:
In the following function "xxx.stream().filter", the value "containsInvalidOptions" will be set.